File: apps/sdkwork-rtc-flutter-mobile/packages/sdkwork_rtc_flutter_mobile_admin_core/lib/src/utils/capability_mapper.dart
const _pluginToBackendCapability = <String, String>{
  'media.audio': 'audio',
  'media.video': 'video',
  'live.broadcast': 'live',
  'live.audience': 'live',
  'screen-share': 'screen-share',
  'recording': 'recording',
  'provider.webhook': 'webhook',
  'provider.active-query': 'active-query',
};

String? mapPluginCapabilityToBackend(String pluginCapability) {
  return _pluginToBackendCapability[pluginCapability];
}

List<String> mapPluginCapabilitiesToBackend(List<String> pluginCapabilities) {
  final mapped = <String>{};
  for (final capability in pluginCapabilities) {
    final backendKey = mapPluginCapabilityToBackend(capability);
    if (backendKey != null) {
      mapped.add(backendKey);
    }
  }
  return mapped.toList();
}

Map<String, bool> profileCapabilitiesToBackendKeys(
  Map<String, dynamic> capabilities,
) {
  return {
    'audio': capabilities['audio'] as bool? ?? false,
    'video': capabilities['video'] as bool? ?? false,
    'live': capabilities['live'] as bool? ?? false,
    'screen-share': capabilities['screenShare'] as bool? ?? false,
    'recording': capabilities['recording'] as bool? ?? false,
    'webhook': capabilities['webhook'] as bool? ?? false,
    'active-query': capabilities['activeQuery'] as bool? ?? false,
  };
}

File: apps/sdkwork-rtc-pc/packages/sdkwork-rtc-pc-admin-core/src/utils/capabilityMapper.ts
const PLUGIN_TO_BACKEND_CAPABILITY: Record<string, string> = {
  "media.audio": "audio",
  "media.video": "video",
  "live.broadcast": "live",
  "live.audience": "live",
  "screen-share": "screen-share",
  recording: "recording",
  "provider.webhook": "webhook",
  "provider.active-query": "active-query",
};

export function mapPluginCapabilityToBackend(pluginCapability: string): string | null {
  return PLUGIN_TO_BACKEND_CAPABILITY[pluginCapability] ?? null;
}

export function mapPluginCapabilitiesToBackend(pluginCapabilities: string[]): string[] {
  const mapped = new Set<string>();
  for (const capability of pluginCapabilities) {
    const backendKey = mapPluginCapabilityToBackend(capability);
    if (backendKey) {
      mapped.add(backendKey);
    }
  }
  return Array.from(mapped);
}

export function profileCapabilitiesToBackendKeys(capabilities: {
  audio: boolean;
  video: boolean;
  live: boolean;
  screenShare: boolean;
  recording: boolean;
  webhook: boolean;
  activeQuery: boolean;
}): Record<string, boolean> {
  return {
    audio: capabilities.audio,
    video: capabilities.video,
    live: capabilities.live,
    "screen-share": capabilities.screenShare,
    recording: capabilities.recording,
    webhook: capabilities.webhook,
    "active-query": capabilities.activeQuery,
  };
}
